.github/workflows/generate-translation-files.yml

```diff
@@ -0,0 +1,137 @@
+# This workflow generates translation files from code and adds the generated
+# translation files to the openedx-translations repository via an auto-merged pull
+# request.
+
+name: Generate Translation Files
+
+on:
+  workflow_dispatch:    # by request
+  schedule:
+    - cron: '0 0 * * *'  # every day at midnight
+
+jobs:
+  setup-branch:
+    runs-on: ubuntu-latest
+    outputs:
+      branch: ${{ steps.dynamic_branch.outputs.branch }}
+    steps:
+    # sets the job output that creates a unique branch name
+    - name: dynamically set branch job output variable
+      id: dynamic_branch
+      run: echo "::set-output name=branch::automated/generate-translations-$(date +'%Y%m%dT%H%M%S')"
+
+    # clones the openedx-translations repo so a branch can be added in the next step
+    - name: clone openedx/openedx-translations
+      uses: actions/checkout@v3
+    
+    # creates the branch where all the translations will be added to
+    - name: make and push a new branch
+      run: |
+        git checkout -b ${{ steps.dynamic_branch.outputs.branch }}
+        git push --set-upstream origin ${{ steps.dynamic_branch.outputs.branch }}
+
+  python-translations:
+    strategy:
+      matrix:
+        repo: [credentials]
+    runs-on: ubuntu-latest
+    needs: [setup-branch]
+
+    steps:
+      # Clones the openedx-translations repo
+      - name: clone openedx/openedx-translations
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.setup-branch.outputs.branch }}
+      
+      # Installs gettext, a dependency for generating translation files in django
+      - name: install gettext
+        run: sudo apt install -y gettext
+
+      # Clones the  repository
+      - name: clone openedx/${{ matrix.repo }}
+        uses: actions/checkout@v3
+        with:
+          repository: openedx/${{ matrix.repo }}
+          path: ${{ matrix.repo }}
+
+      # Sets up Python
+      - name: setup python
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.8'
+
+      # Installs Python requirements from translations.txt
+      - name: install requirements
+        run: |
+          cd ${{ matrix.repo }}
+          pip install -r requirements/translations.txt
+
+      # Generates the english translation files
+      - name: generate translations
+        run: |
+          cd ${{ matrix.repo }}
+          make extract_translations
+      
+      # Add only the english translation files
+      - name: add the english translation files
+        run: |
+          # set identity
+          git config --global user.email "cmuniz@tcril.org"
+          git config --global user.name "Carlos Muniz"
+          # finds the directory containing the english locale usually located in
+          # */*/conf/locale/en
+          EN_DIR=$(find . -type d -name "en")
+          # remove ${{ matrix.repo }}/.git so we don't commit a submodule
```

Comment:
Is this an issue that you ran into? It seems weird on the surface, but I'm quite possibly missing something important here.

Reply:
Yes. I ran into the issue of trying to add only the english django.po and djangojs.po files, but because a repo with the .git directory was inside the repo, it wouldn't add the translation files. If I instead added the credentials directory, it would add the whole directory as a submodule repository, which is not what we want. By deleting the .git dir, credentials is just a normal dir that I can use find in, and add translation files from.